base_layer/wallet/src/key_manager_service/storage/sqlite_db/mod.rs

```diff
@@ -115,8 +111,8 @@ impl KeyManagerBackend for KeyManagerSqliteDatabase {
         let conn = self.database_connection.get_pooled_connection()?;
         let acquire_lock = start.elapsed();
 
-        let mut km_sql = NewKeyManagerStateSql::from(key_manager);
-        self.encrypt_if_necessary(&mut km_sql)?;
+        let km_sql = NewKeyManagerStateSql::from(key_manager);
```

Comment:
I think it's better here to have the SQL structs always encrypted, and to pass the cipher, or create the cipher in the from
```suggestion
        let km_sql = NewKeyManagerStateSql::from(key_manager, cipher);
```

Reply:
Refactored accordingly